### ISSUE TITLE
Output window sub-tabs don't honor TabbedPane foreground color settings

### ISSUE THREAD

**Issue by rkeen-siemens** [apache/netbeans#5098]
### Apache NetBeans version

Apache NetBeans 16

### What happened

Attempting to modify the tabbed pane foreground colors does not work for output window sub-tabs.

### How to reproduce

Go to Tools > Options > Appearance > FlatLaf and click Edit custom properties and add the following line

```
TabbedPane.selectedForeground = $TabbedPane.underlineColor
```

Restart the IDE and open two output window sub-tabs (e.g. by running two files from different projects). I would expect the selected tab to be the same as the focused underline color. If you make the same change for `EditorTab.selectedForeground`, those tabs do render with the correct color.

### Did this work correctly in an earlier version?

No / Don't know

### Operating System

Windows 10 Pro, CentOS 7.6 with Gnome 3

### JDK

Java 17.0.4.1

### Apache NetBeans packaging

Apache NetBeans provided installer

### Anything else

These tabs appear to be generated via `org.netbeans.core.windows.view.ui.CloseButtonTabbedPane` which sets a custom component to render the title for each tab (`ButtonTab`). This component should inherit the colors from the enclosing tabbed pane or provide a separate setting to allow setting the colors as done with `EditorTab` and `ViewTab`.

### Are you willing to submit a pull request?

No

### Code of Conduct

Yes

**Comment by neilcsmith-net** [comment]
Makes sense IMO, and the label should probably just inherit the container colour.  Could possibly provide extra key, defaulting to tabbed pane colour, for the outer container.

Aside - amount of LAF specific code in that class is hideous.

**Comment by IanKrL** [comment]
These tabs also don't honor font settings. If you set:
`TabbedPane.font = +bold`
Nothing happens to the Output window sub-tabs.

My fix is to add
`label.setFont(UIManager.getFont("TabbedPane.font"));`
in the ButtonTab constructor, after 'label' has been declared.

**Comment by neilcsmith-net** [comment]
Given FlatLaf has its own closable tabs (eg. https://github.com/JFormDesigner/FlatLaf/pull/193 ) might also want to see if we should provide an alternative `TabbedPaneFactory`, or otherwise different implementation of `CloseButtonTabbedPane` when it is the active LaF? cc/ @DevCharly - what do you think?

**Comment by rkeen-siemens** [comment]
That sounds like a good idea and perhaps also use that factory for the top level tabs as well (editor and view tabs).
